server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java

```diff
@@ -166,6 +171,13 @@ private class EventHandler implements Runnable {
          * A condition variable for waking up the event handler thread.
          */
         private final Condition cond = lock.newCondition();
+        private final Consumer<Integer> onQueueSizeChange;
+
+        private AtomicInteger size = new AtomicInteger(0);
```

Comment:
Long? Is there a reason we are limiting this to an integer?

Reply:
There should never be that many events in the queue that can't fit in an int. Each event is a runnable, held in a linked list. Mind that `java.util.LinkedList` for example, uses `int` for size.